frontend/src/app/coworking/coworking.models.ts

```diff
@@ -38,6 +38,17 @@ export interface Room {
   nickname: string;
 }
 
+export interface RoomDetails {
+  id: string | null;
+  nickname: string;
+  building: string;
+  room: string;
+  capacity: number;
+  reservable: boolean;
+  description: string;
+  seats: Seat[];
+}
+
```

Comment:
Typically, the frontend does not use `Details` models because dealing with types becomes more difficult - we instead have used `| null` on all of the fields that would be part of `Details` not part of the original model.

If we must use a details model here, make it extend `Room`!

Reply:
Looping in @aarjavjain2002